tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
So... by "access" this means *write*? Also, I am not aware of any conditions for inserting `StorageLive`, this is suspicious.